sources/BigInt.swift

```diff
@@ -259,6 +259,17 @@ extension BigInt: IntegerArithmetic {
     public static func %(a: BigInt, b: BigInt) -> BigInt {
         return BigInt(abs: a.abs % b.abs, negative: a.negative)
     }
+  
+    /// Return the result of `a` mod `b`.
+    public static func modulus(a: BigInt, b: BigInt) -> BigInt {
+        if a.negative == b.negative {
+            return BigInt(abs: a.abs % b.abs, negative: a.negative)
+        }
+        else {
+            let floor = ((a / b) - 1) * b
+            return BigInt(abs: (floor.abs - a.abs) % b.abs, negative: b.negative)
```

Comment:
Oh, I wasn't aware of Mathematica's modulo. Industry practice [seems to be that everyone does whatever they like][wiki], and I like the modulo being nonnegative, so let's get this merged! :)

[wiki]: https://en.m.wikipedia.org/wiki/Modulo_operation